flake8_deprecated.py

```diff
@@ -1,49 +1,40 @@
 # -*- coding: utf-8 -*-
-try:
-    from flake8.engine import pep8 as stdin_utils
-except ImportError:
-    from flake8 import utils as stdin_utils
+import ast
 
 
 class Flake8Deprecated(object):
     name = 'flake8_deprecated'
     version = '1.2'
     message = 'D001 found {0:s} replace it with {1:s}'
     checks = {
-        'assertEqual': ('failUnlessEqual(', 'assertEquals(', ),
-        'assertNotEqual': ('failIfEqual(', ),
-        'assertTrue': ('failUnless(', 'assert_(', ),
-        'assertFalse': ('failIf(', ),
-        'assertRaises': ('failUnlessRaises(', ),
-        'assertAlmostEqual': ('failUnlessAlmostEqual(', ),
-        'assertNotAlmostEqual': ('failIfAlmostEqual(', ),
-        'AccessControl.ClassSecurityInfo.protected': ('declareProtected(', ),
-        'AccessControl.ClassSecurityInfo.private': ('declarePrivate(', ),
-        'AccessControl.ClassSecurityInfo.public': ('declarePublic(', ),
-        'zope.interface.provider': ('directlyProvides(', ),
-        'zope.interface.implementer': ('classImplements(', ),
-        'self.loadZCML(': ('xmlconfig.file(', ),
-        'zope.interface.implementer': ('implements(', ),
-        'zope.component.adapter': ('adapts(', ),
+        'assertEqual': ('failUnlessEqual', 'assertEquals', ),
+        'assertNotEqual': ('failIfEqual', ),
+        'assertTrue': ('failUnless', 'assert_', ),
+        'assertFalse': ('failIf', ),
+        'assertRaises': ('failUnlessRaises', ),
+        'assertAlmostEqual': ('failUnlessAlmostEqual', ),
+        'assertNotAlmostEqual': ('failIfAlmostEqual', ),
+        'AccessControl.ClassSecurityInfo.protected': ('declareProtected', ),
+        'AccessControl.ClassSecurityInfo.private': ('declarePrivate', ),
+        'AccessControl.ClassSecurityInfo.public': ('declarePublic', ),
+        'zope.interface.provider': ('directlyProvides', ),
+        'zope.interface.implementer': ('classImplements', ),
+        'self.loadZCML(': ('xmlconfig.file', ),
+        'zope.interface.implementer': ('implements', ),
+        'zope.component.adapter': ('adapts', ),
     }
 
-    def __init__(self, tree, filename):
-        self.filename = filename
+    def __init__(self, tree):
         self.flat_checks = self._flatten_checks()
+        self.tree = tree
 
     def run(self):
-        if self.filename == 'stdin':
-            lines = stdin_utils.stdin_get_value().splitlines(True)
-        else:
-            with open(self.filename, 'rb') as f:
-                lines = [line.decode('utf-8') for line in f.readlines()]
```

Comment:
@alexmuller what was the reason to ignore this 'stdin' if conditional? @iham added it to support using this plugin in combination with text editors.

Using only the tree and ignoring the filename guarantees that this combination will keep working? I've never tested it myself to be honest.